bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/ICalendarHandler.java

```diff
@@ -0,0 +1,396 @@
+    }
+
+    private void executeEventCommands(@Nullable List<Event> events, CommandTagType execTime) {
+        // no begun or ended events => exit quietly as there is nothing to do
+        if ((events == null) || (events.isEmpty())) {
+            return;
+        }
+        // prevent synchronization issues (MVN null pointer warnings) in "eventPublisherCallback"
+        @Nullable
+        EventPublisher syncEventPublisherCallback = eventPublisherCallback;
+        if (syncEventPublisherCallback == null) {
+            logger.warn("EventPublisher object not instantiated!");
+            return;
+        }
+        // prevent potential synchronization issues (MVN null pointer warnings) in "configuration"
+        @Nullable
+        ICalendarConfiguration syncConfiguration = configuration;
+        if (syncConfiguration == null) {
+            logger.warn("Configuration not instantiated!");
+            return;
+        }
+        // loop through all events in the list
+        for (Event event : events) {
+
+            // loop through all command tags in the event
+            for (CommandTag cmdTag : event.commandTags) {
+
+                // only process the BEGIN resp. END tags
+                if (cmdTag.getTagType() != execTime) {
+                    continue;
+                }
+                if (!cmdTag.isAuthorized(syncConfiguration.authorizationCode)) {
+                    logger.warn("Event: {}, Command Tag: {} => Not authorized!", event.title, cmdTag.getFullTag());
+                    continue;
+                }
+                if (!cmdTag.getItemName().matches("^\\w+$")) {
+                    logger.warn("Event: {}, Command Tag: {} => Bad syntax for Item name!", event.title,
+                            cmdTag.getFullTag());
+                    continue;
+                }
+                Command cmdState = cmdTag.getCommand();
+                if (cmdState == null) {
+                    logger.warn("Event: {}, Command Tag: {} => Invalid Target State!", event.title,
+                            cmdTag.getFullTag());
+                    continue;
+                }
+
+                // (try to) execute the command
+                try {
+                    syncEventPublisherCallback
+                            .post(ItemEventFactory.createCommandEvent(cmdTag.getItemName(), cmdState));
+                    if (logger.isDebugEnabled()) {
+                        String cmdType = cmdState.getClass().toString();
+                        int index = cmdType.lastIndexOf(".") + 1;
+                        if ((index > 0) && (index < cmdType.length())) {
+                            cmdType = cmdType.substring(index);
+                        }
+                        logger.debug("Event: {}, Command Tag: {} => {}.postUpdate({}: {})", event.title,
+                                cmdTag.getFullTag(), cmdTag.getItemName(), cmdType, cmdState);
+                    }
+                } catch (IllegalArgumentException e) {
```

Comment:
It's not fully preventable as it may be created by the framework if the eventbus is not available or if the command would be null (which is not the case). However i've made the error message more understandable and added a stacktrace to debug - as this is maybe a situation where people may be happy about additional debug output.